debugutils/resources.go

```diff
@@ -0,0 +1,213 @@
+package debugutils
+
+import (
+	"encoding/json"
+	"fmt"
+	"os"
+	"path/filepath"
+	"sync"
+
+	"github.com/solo-io/go-utils/errors"
+	"github.com/solo-io/go-utils/installutils/helmchart"
+	"github.com/solo-io/go-utils/installutils/kuberesource"
+	"github.com/solo-io/go-utils/kubeutils"
+	"github.com/spf13/afero"
+	"golang.org/x/sync/errgroup"
+	apiextensions "k8s.io/apiextensions-apiserver/pkg/apis/apiextensions/v1beta1"
+	"k8s.io/apimachinery/pkg/api/meta"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+	"k8s.io/apimachinery/pkg/runtime/schema"
+	"k8s.io/client-go/dynamic"
+	"sigs.k8s.io/controller-runtime/pkg/client/apiutil"
+)
+
+const (
+	resourceCollectorStr = "resourceCollector"
+)
+
+var (
+	initializationError = func(err error, obj string) error {
```

Comment:
nit: since this is used across a bunch of abstractions probably makes sense to just go in an errors.go file